prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt

```diff
@@ -274,8 +275,11 @@ class HL7Reader(private val actionLogger: ActionLogger) : Logging {
             rawHL7: String,
             parseConfiguration: HL7MessageParseAndConvertConfiguration?,
         ): Message {
-            val hl7MessageType = getMessageType(rawHL7)
-            return getHL7ParsingContext(hl7MessageType, parseConfiguration).pipeParser.parse(rawHL7)
+            // A carriage return if the official segment delimiter and a new line is not recognized so we replace
```

Comment:
Soo.....the reason this isn't an issue in the tests for fhir converter is that the code uses the iterator which does this before even trying to do any parsing.  I've added here even though it's redundant so that it remains safe to use in other situations (like the translation tests)

Reply:
How well does this work for files edited in Windows? Those files typically end lines with `\r\n` which for this code would translate to `\r\r`

Reply:
good call